api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto

```diff
@@ -148,4 +148,8 @@ message TcpProxy {
   // The maximum number of unsuccessful connection attempts that will be made before
   // giving up. If the parameter is not specified, 1 connection attempt will be made.
   google.protobuf.UInt32Value max_connect_attempts = 7 [(validate.rules).uint32 = {gte: 1}];
+
+  // Optional configuration for TCP proxy hash. Source IP will be used for selecting hosts when the
+  // parameter is set to true.
+  bool hash_with_source_ip = 11;
```

Comment:
Should this be a `oneof`? Will we want other kind of hash selection for TCP proxy? E.g. 5-tuple ECMP?

Reply:
I think we might need to support hash based on 5-tuple and we should provide `and` logic for 5-tuple instead of `or`, e.g. hash based on source ip and source port.

Reply:
Just to clarify there though, IMO it would be a lot more future proof here if we went with a oneof approach with a required hashing policy. Can we move to that pattern so avoid costly refactors later? Is there any reason not to do that?

Reply:
```
  // Optional configuration for TCP proxy hash policy.
  oneof hash_policy {
    // The source IP will be used to compute the hash used by hash-based load balancing algorithms.
    bool source_ip = 11;
  }
```
Would this proto file look good to you? Should I add a explicit `none` entry in the hash_policy or allow hash_policy to be optional?

Reply:
Yeah that looks about right. What I would probably do is just put the oneof inside a message called `HashPolicy`. Then the `HashPolicy` itself can be optional. If it is specified then I would make the oneof itself required.

Reply:
Updated the API and add hash policy for TCP proxy. PTAL